legend-engine-xt-graphQL-pure/src/main/resources/core_external_query_graphql/binding/fromPure/introspection/fromPure_Introspection.pure

```diff
@@ -5,13 +5,24 @@ import meta::external::query::graphQL::binding::fromPure::introspection::*;
 
 function meta::external::query::graphQL::binding::fromPure::introspection::scanTypes(t:Type[1], processed:Type[*]):Type[*]
 {
-   if (!$processed->contains($t),
-       |$t->concatenate(if($t->instanceOf(Class),
-                         |$t->cast(@Class<Any>)->allProperties().genericType.rawType->distinct()->concatenate($t->cast(@Class<Any>)->meta::pure::functions::meta::findAllSpecializations())->filter(t|($t->instanceOf(Class) || $t->instanceOf(meta::pure::metamodel::type::Enumeration)) && $t != Any)->map(c|$c->meta::external::query::graphQL::binding::fromPure::introspection::scanTypes($processed->concatenate($t)))->distinct(),
-                         |[]
-                       )),
-       |[]
-   );
+  let np = if(!$processed->contains($t), | $processed->concatenate($t), | $processed);
+  if (!$processed->contains($t) && $t->instanceOf(Class),
```

Comment:
Would swapping the conjuncts in this condition make this more efficient? If you check the instance of $t first before doing the contains check you avoid having to go through processed if you don't need to. Might make a difference if the size of $processed gets very large.

Reply:
yes doing the contains only once is better